src/unread/__tests__/unreadModel-test.js

```diff
@@ -212,9 +220,38 @@ describe('stream substate', () => {
       expect(newState.streams.get(234)).toBe(state.streams.get(234));
     });
 
-    test('when operation is "remove" do nothing', () => {
-      const action = mkAction({ messages: [1, 2], op: 'remove' });
-      expect(reducer(baseState, action, eg.plusReduxState)).toBe(baseState);
+    test('when operation is "remove", add stream messages to unreads', () => {
+      const action = mkAction({
+        messages: [42, 99],
+        op: 'remove',
+        message_info_dict: new Map([
+          [
+            42,
+            {
+              type: 'stream',
+              mentioned: false,
+              sender_id: eg.selfUser.user_id,
+              stream_id: 123,
+              topic: 'foo',
+              unmuted_stream_msg: true,
+            },
```

Comment:
Let's put this in a little helper (perhaps even local to this test case), taking just `stream_id` and `topic` as arguments. Then all this mess only appears once, and this test data gets a lot more compact -- like one line per message. That helps a lot in reading through these tests to see what kinds of input they're covering.